clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkThread.java

```diff
@@ -84,6 +90,8 @@ public ConsumerNetworkThread(LogContext logContext,
         this.networkClientDelegateSupplier = networkClientDelegateSupplier;
         this.requestManagersSupplier = requestManagersSupplier;
         this.running = true;
+        this.metricsManager = metricsManager;
+        this.usingMetrics = (metricsManager != null);
```

Comment:
i'm not sure why we need a variable like this.  Make the metricsManager optional if it is not always there.

Reply:
Yes, `Optional<NetworkThreadMetricsManager>` is neater.